dev-tools/packer/xgo-scripts/before_build.sh

```diff
@@ -1,9 +1,35 @@
 #!/bin/sh
 
-PB_PATH=/go/src/github.com/elastic/packetbeat
+set -e
 
-if [ -d $PB_PATH ]; then
-	cd $PB_PATH
+BEATS_PATH=/go/src/github.com/elastic/beats
+
+# BEATNAME is in the $PACK variable
+BEATNAME=$PACK
+
+if [ $BEATNAME = "packetbeat" ]; then
+	cd $BEATS_PATH
 	patch -p1 < /gopacket_pcap.patch
-    PREFIX=/build make install-cfg
 fi
+
+
+BEAT_PATH=$BEATS_PATH/$BEATNAME
+cd $BEAT_PATH
+
+PREFIX=/build
+
+echo $PREFIX
+
+cp etc/$BEATNAME.template.json $PREFIX/$BEATNAME.template.json
+# linux
+cp $BEATNAME.yml $PREFIX/$BEATNAME-linux.yml
+# binary
+cp $BEATNAME.yml $PREFIX/$BEATNAME-binary.yml
+# darwin
+cp $BEATNAME.yml $PREFIX/$BEATNAME-darwin.yml
+# win
+cp $BEATNAME.yml $PREFIX/$BEATNAME-win.yml
+
+
+# Contains beat specific adjustments. As it is platform specific knowledge, it should be in packer not the beats itself
+PREFIX=$PREFIX make adjust-cfg
```

Comment:
Additional note: I would hope that the platform specific things get obsolete when we have a config and data directory.